tools/approval-service/internal/approvalservice/eventprocessor.go

```diff
@@ -0,0 +1,169 @@
+	for _, v := range p.validation {
+		for _, env := range v.Environments {
+			env, err := p.githubClient.GetEnvironment(context.TODO(), github.GetEnvironmentInfo{
+				Org:         v.Org,
+				Repo:        v.Repo,
+				Environment: env,
+			})
+			if err != nil {
+				return fmt.Errorf("getting environment %q: %w", env, err)
+			}
+			p.envNameToID[envEntry(v.Org, v.Repo, env.Name)] = env.ID
+		}
+	}
+
+	return nil
+}
+
+func (p *processor) ProcessDeploymentReviewEvent(e githubevents.DeploymentReviewEvent, valid bool) error {
+	if !valid {
+		// TODO: Create a rejected access request if the event is invalid (e.g. incorrect org, env, user, etc.)
+		//       This should be useful for audit purposes.
+		return fmt.Errorf("invalid event")
+	}
+
+	// Create access request
+	// The name of an access request must be a UUID
+	name := uuid.New().String()
+	p.log.Info("Creating access request", "access_request_name", name)
+	req, err := types.NewAccessRequest(name, p.TeleportUser, p.TeleportRole)
+	if err != nil {
+		return fmt.Errorf("generating access request: %w", err)
+	}
+
+	req.SetRequestReason("Deployment review for environment " + e.Environment)
+	req.SetStaticLabels(map[string]string{
+		"workflow_run_id": strconv.Itoa(int(e.WorkflowID)),
+		"organization":    e.Organization,
+		"repository":      e.Repository,
+		"environment":     e.Environment,
+	})
+	created, err := p.teleportClient.CreateAccessRequestV2(context.TODO(), req)
+	if err != nil {
+		return fmt.Errorf("creating access request %q: %w", name, err)
+	}
+
+	if err != nil {
+		return fmt.Errorf("creating access request %q: %w", name, err)
+	}
+	p.log.Info("Created access request", "access_request_name", created.GetName())
+	return nil
+}
+
+func (p *processor) HandleReview(ctx context.Context, req types.AccessRequest) error {
+	p.log.Info("Handling review", "access_request_name", req.GetName())
+
+	runIDLabel := req.GetStaticLabels()["workflow_run_id"]
+	if runIDLabel == "" {
+		return fmt.Errorf("missing workflow run ID in access request %s", req.GetName())
+	}
+
+	orgLabel := req.GetStaticLabels()["organization"]
+	if orgLabel == "" {
+		return fmt.Errorf("missing organization in access request %s", req.GetName())
+	}
+
+	repoLabel := req.GetStaticLabels()["repository"]
+	if repoLabel == "" {
+		return fmt.Errorf("missing repository in access request %s", req.GetName())
+	}
+
+	env := req.GetStaticLabels()["environment"]
+	if env == "" {
+		return fmt.Errorf("missing environment in access request %s", req.GetName())
+	}
+
+	fmt.Printf("is map: %+v", p.envNameToID)
+	envID, ok := p.envNameToID[envEntry(orgLabel, repoLabel, env)]
+	if !ok {
+		return fmt.Errorf("encountered unkown environment %q, this indicates a problem during setup", envEntry(orgLabel, repoLabel, env))
+	}
+
+	runID, err := strconv.Atoi(runIDLabel)
+	if err != nil {
+		return fmt.Errorf("parsing workflow run ID: %w", err)
+	}
+
+	state := github.PendingDeploymentApprovalState_APPROVED
```

Comment:
In the off chance that more states are added in the future, this should default to rejected, with the state set to "approved" if the request is approved. Alternatively this could implement a `switch` statement with a default case of denial.